themes/classic/_dev/js/product.js

```diff
@@ -104,11 +104,16 @@ $(document).ready(function () {
       max: 1000000
     });
 
-    quantityInput.on('change', function (event) {
-      let $productRefresh = $('.product-refresh');
-      $(event.currentTarget).trigger('touchspin.stopspin');
-      $productRefresh.trigger('click', {eventType: 'updatedProductQuantity'});
-      event.preventDefault();
+    var quantity = quantityInput.val();
+    quantityInput.on('keyup change', function (event) {
+      const newQuantity = $(this).val();
+      if (newQuantity !== quantity) {
+        quantity = newQuantity;
+        let $productRefresh = $('.product-refresh');
+        $(event.currentTarget).trigger('touchspin.stopspin');
+        $productRefresh.trigger('click', {eventType: 'updatedProductQuantity'});
+        event.preventDefault();
```

Comment:
This should always be done, so I think it should be moved outside the `if` block